pkg/controller/v1alpha2/experiment/experiment_controller.go

```diff
@@ -368,6 +368,26 @@ func (r *ReconcileExperiment) createTrials(instance *experimentsv1alpha2.Experim
 }
 
 func (r *ReconcileExperiment) deleteTrials(instance *experimentsv1alpha2.Experiment, deleteCount int32) error {
+	logger := log.WithValues("Experiment", types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()})
+	trials := &trialsv1alpha2.TrialList{}
+	labels := map[string]string{consts.LabelExperimentName: instance.Name}
+	lo := &client.ListOptions{}
+	lo.MatchingLabels(labels).InNamespace(instance.Namespace)
 
+	if err := r.List(context.TODO(), lo, trials); err != nil {
+		logger.Error(err, "Trial List error")
+		return err
+	}
+	actualDeleteCount := int(deleteCount)
+	if int32(len(trials.Items)) < deleteCount {
+		actualDeleteCount = len(trials.Items)
+	}
+
+	for i := 0; i < actualDeleteCount; i++ {
+		if err := r.Delete(context.TODO(), &trials.Items[len(trials.Items)-1-i]); err != nil {
```

Comment:
We have to delete the most recently created trial. Is the last item of the list guaranteed to be the latest trial?

Reply:
During my test, it is. But I think I should make sure of it. Please wait a while